File: split/endname.pas
procedure endname;
begin
  if strptr + 3 > maxstrings then
    overflow(258, maxstrings - initstrptr);
  if areadelimiter = 0 then
    curarea := 338
  else
  begin
    curarea := strptr;
    strstart[strptr + 1] := strstart[strptr] + areadelimiter;
    strptr := strptr + 1;
  end;
  if extdelimiter = 0 then
  begin
    curext := 338;
    curname := makestring;
  end
  else
  begin
    curname := strptr;
    strstart[strptr + 1] := strstart[strptr] + extdelimiter - areadelimiter - 1;
    strptr := strptr + 1;
    curext := makestring;
  end;
end;

File: cpp/src/endname.cpp
#include "endname.h"
#include "erreur.h"
#include "makestring.h"
#include "texte.h"

void endname(void)
{
	curarea = currentString.substr(0, areadelimiter);
	if (areadelimiter)
		strings.push_back(curarea);
	if (extdelimiter == 0)
	{
		curext = "";
		curname = currentString.substr(areadelimiter);
	}
	else
	{
		curext = currentString.substr(extdelimiter-1);
		curname = currentString.substr(areadelimiter, extdelimiter-1);
	}
	strings.push_back(curname);
	if (extdelimiter)
		strings.push_back(curext);
}
